spark/src/main/scala/io/delta/sharing/spark/util/RetryUtils.scala

```diff
@@ -71,6 +71,9 @@ private[sharing] object RetryUtils extends Logging {
           false
         }
       case _: java.net.SocketTimeoutException => true
+      // do not retry on ConnectionClosedException because it can be caused by invalid json returned
+      // from the delta sharing server.
+      case _: org.apache.http.ConnectionClosedException => false
```

Comment:
Are there other cases where this should be retried (for example, some networking error) which may be more common?

Reply:
that's a good question, I'm not sure about the side effect of this change. As it's the client side change, we don't have good logs/stats on what are all the cases that trigger this error. 

let me see if the current code works with a server side change, if not we can add more detailed check here but may still need it in some way.

Reply:
SG.